docs/guides/jupyter.rst

```diff
@@ -0,0 +1,398 @@
+  to reduce repository size and improve integration with git.
+
+Background
+----------
+
+.. _nb-background:
+
+Existing relevant extensions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are two main extensions that add support Jupyter notebooks as source files in Sphinx:
+nbsphinx_ and MyST-NB_. They have similar intent and basic functionality:
+both can read notebooks in ``.ipynb`` and additional formats supported by `jupytext`_,
+and are configured in a similar way.
+However, there are some differences between them:
+
+- nsphinx uses `pandoc <https://pandoc.org/>`_
+  to convert the Markdown from Jupyter notebooks to reStructuredText
+  and then to `docutils AST <https://docutils.sourceforge.io/docs/ref/doctree.html>`_,
+  whereas MyST-NB uses `MyST-Parser`_
+  to directly convert the Markdown text to docutils AST.
+  Therefore, nbsphinx assumes `pandoc flavored Markdown <https://pandoc.org/MANUAL.html#pandocs-markdown>`_,
+  whereas MyST-NB uses `MyST flavored Markdown <https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html>`_.
+  Both Markdown flavors are mostly equal,
+  but they have some differences.
+- nbsphinx executes each notebook during the parsing phase,
+  whereas MyST-NB can execute all notebooks up front
+  and cache them with `jupyter-cache <https://jupyter-cache.readthedocs.io/>`_.
+  This can result in shorter build times when when notebooks are modified
+  if using MyST-NB.
+- nbsphinx provides functionality to create thumbnail galleries,
+  whereas MyST-NB does not have such functionality at the moment
+  (see :ref:`notebook-galleries` for more information about galleries).
+- MyST-NB allows embedding Python objects coming from the notebook in the documentation
+  (read :ref:`their "glue" documentation <myst-nb:glue>`
+  for more information)
+  and provides more sophisticated :ref:`error reporting <myst-nb:start/error-reporting>`
+  than the one nbsphinx has.
+- The visual appearance of code cells and their outputs is slightly different:
+  nbsphinx renders the cell numbers by default,
+  whereas MyST-NB doesn't.
+
+.. _nbsphinx: https://nbsphinx.readthedocs.io/
+.. _MyST-NB: https://myst-nb.readthedocs.io/
+.. _MyST-Parser: https://myst-parser.readthedocs.io/
+.. _jupytext: https://jupytext.readthedocs.io/
+
+Using which one to use depends on your use case. As general recommendations:
+
+- If you want to use :ref:`other notebook formats <other-formats>`
+  or :ref:`generate a thumbnail gallery from your notebooks <notebook-galleries>`,
+  nbsphinx is the right choice.
+- If you want to leverage a more optimized execution workflow
+  and a more streamlined parsing mechanism,
+  as well as some of the unique MyST-NB functionalities
+  (``glue``, better error reporting)
+  you should use MyST-NB.
+
+For the rest of this document we will focus on `nbsphinx`_,
+and we will point out the differences with `MyST-NB`_ where appropriate.
+Notice that they can't both be used at the same time.
```

Comment:
It seems we can delete this paragraph, the guide is about to finish and we already presented both options.

Reply:
Good catch!